Algebra.java

```diff
@@ -6,62 +6,192 @@
 public class Algebra {
```

Comment:
The amount of HW violations in this assignment is staggering..
I will clarify again: during the assignment, you are allowed to use only things you were taught in Lectures or Recitations, unless you were specifically told otherwise.

Each of those violation cause deduction of points..
Total deductions: 14 points.

The logic was questionable at best in Anagram.
Talk to me,
Tal Danai